library/alloc/src/vec.rs

```diff
@@ -116,8 +116,10 @@ use crate::raw_vec::RawVec;
 /// assert_eq!(vec, [0, 0, 0, 0, 0]);
 ///
 /// // The following is equivalent, but potentially slower:
-/// let mut vec1 = Vec::with_capacity(5);
-/// vec1.resize(5, 0);
+/// let mut vec = Vec::with_capacity(5);
+/// assert_eq!(vec, []);
```

Comment:
I think this assert is a little redundant, `with_capacity` already has an example like this and it interrupts the flow of the example.
```suggestion
```

Reply:
I understand but this is just to clarify to the readers. This section appears before `with_capacity` or any discussion related to capacity so we assumed the reader does not know what is `with_capacity`. It does disrupt the flow a bit even I was confused what `resize` does so just to confirm I add this.

Reply:
Hmm, instead of adding the assert could we link to the capacity section? I think that would be more helpful.

Reply:
`#capacity-and-reallocation` do you have any idea where should we put that?

Reply:
Maybe `performing allocation and initialization in separate steps` could be a link? Or just `performing allocation` if that's too long?

Reply:
I will just commit this, maybe you can follow up on that.